suzieq/engines/pandas/interfaces.py

```diff
@@ -89,6 +87,69 @@ def get(self, **kwargs):
         else:
             return df.reset_index(drop=True)[fields]
 
+    def _check_vlan_match(self, vlan_list: List[str],
+                          df: pd.DataFrame) -> pd.DataFrame:
+        '''Return a dataframe with rows in VLANs requested
+
+        VLAN is treated specially because its the only field that is a list
+        of integers, and can be filtered on multiple fields: vlan and vlanList
+        A user filter on VLAN is a list that can contain numeric comparisons
+        as well as simple integers.
+
+        Its not made into a generic filter match because this match on VLAN
+        applies only to interface table.
+        '''
+
+        # If there are any VLANs with ! mixed without !, remove the
+        # ! ones. vlanList is an OR list, not an AND list, except
+        # if all are !
+        if any(x.startswith('!') for x in vlan_list):
+            cond = 'and'
+        else:
+            cond = 'or'
+
+        resdf_list = []
+        if (any(x.startswith('>') for x in vlan_list) and
+                any(x.startswith('<') for x in vlan_list)):
+            # if the user specifies both >/>= and </<=, then we
+            # treat this as an and
+            cond = 'and'
+
+        opdict = {'<': operator.lt, '>': operator.gt,
+                  '<=': operator.le, '>=': operator.ge,
+                  '!': operator.ne, '==': operator.eq}
+
+        for vlan in vlan_list:
+            if vlan.startswith(('<=', '>=')):
+                op = vlan[0:2]
+                vlan = vlan[2:].strip()
+            elif vlan.startswith(('<', '>', '!')):
+                op = vlan[0]
+                vlan = vlan[1:].strip()
+            else:
+                op = '=='
+
+            vlan = int(vlan)
+            if op == "!":
+                tmpdf = df[df.apply(
+                    lambda x, vlan: all(opdict[op](y, vlan)
+                                        for y in x.vlanList) and
+                    opdict[op](x.vlan, vlan), args=(vlan,), axis=1)]
+            else:
+                tmpdf = df[df.apply(
+                    lambda x, vlan: any(opdict[op](y, vlan)
```

Comment:
If we directly assign the operation to `op` we can directly do `op(..)`, which I think is more readeable

Reply:
Was trying to avoid using up another variable when I use it only once. Its not that unreadable